src/app/modules/payments/payments.service.ts

```diff
@@ -330,3 +337,70 @@ export const findLatestSuccessfulPaymentByEmailAndFormId = (
     return okAsync(result)
   })
 }
+
+/**
+ * Retrieves payments that are in PaymentStatus.Pending satisfying X hours agos
+ * @param createdHrsAgo defaults to 72
+ * @returns
+ */
+
+export const findPendingPaymentsByTime = (createdHrsAgo = 72) => {
+  const logMeta = {
+    action: 'findPaymentByTime',
+    createdHrsAgo,
+  }
+  const positiveHrsAgo = Math.max(createdHrsAgo, 1)
```

Comment:
Didn't we say we wanted to verify with 30mins interval? the code here restrict to minimum 1 hour.

Reply:
Resolved, since cron will run every 30mins, and try to sync all incomplete payments.